adyenv6b2ccheckoutaddon/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java

```diff
@@ -534,16 +540,47 @@ public void handlePaymentForm(AdyenPaymentForm adyenPaymentForm, BindingResult b
             return;
         }
 
+        CustomerModel customer = handleGuestCustomer(cartModel);
+
         //Update CartModel
         cartModel.setAdyenCseToken(adyenPaymentForm.getCseToken());
         cartModel.setAdyenDfValue(adyenPaymentForm.getDfValue());
+        cartModel.setUser(customer);
+        recalculateCart(cartModel);
```

Comment:
If the customer on cart is changed the cart needs to be recalculated. When not, order cannot be placed as isCalculated=FALSE.